File: admin/src/main/java/com/jigumulmi/banner/dto/response/AdminBannerResponseDto.java
package com.jigumulmi.banner.dto.response;

import com.jigumulmi.banner.domain.Banner;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.experimental.SuperBuilder;

@Getter
@SuperBuilder
public class AdminBannerResponseDto {

    private LocalDateTime modifiedAt;
    private Long id;
    private String title;
    private Boolean isActive;

    public static AdminBannerResponseDto from(Banner banner) {
        return AdminBannerResponseDto.builder()
            .id(banner.getId())
            .title(banner.getTitle())
            .modifiedAt(banner.getModifiedAt())
            .isActive(banner.getIsActive())
            .build();
    }
}

File: admin/src/main/kotlin/com/jigumulmi/banner/dto/response/AdminBannerResponseDto.kt
package com.jigumulmi.banner.dto.response

import com.jigumulmi.banner.domain.Banner
import java.time.LocalDateTime

data class AdminBannerResponseDto(
    val modifiedAt: LocalDateTime,
    val id: Long,
    val title: String? = null,
    val isActive: Boolean? = false,
    ) {

    companion object {
        fun from(banner: Banner): AdminBannerResponseDto {
            return AdminBannerResponseDto(
                id = banner.id,
                title = banner.title,
                isActive = banner.isActive,
                modifiedAt = banner.modifiedAt,
            )
        }
    }
}
